src/py/gee/utils.py

```diff
@@ -404,16 +404,20 @@ def reduceRegion(image):
         geometry = ee.Geometry.Point(coords)
 
     if sourceName.lower() == 'landsat':
-        collection = getLandsatToa(startDate, endDate, geometry)
+        collection = (getLandsatToa(startDate, endDate, geometry)
+                      .select([indexName],['index']))
     elif sourceName.lower() == 'nicfi':
         collection = getNICFI({
             'start':startDate,
             'end':endDate
-        })
+        }).select([indexName],['index'])
+    elif sourceName.lower() == "custom":
+        collection = (ee.ImageCollection(assetId)
+                      .filterBounds(geometry).filterDate(startDate, endDate)).select([band],['index'])
     else:
         raise ValueError(f'imagery source {sourceName} is not implemented.') 
-
-    collection = collection.select([indexName],['index'])
+   
+    # collection = collection.select([indexName],['index']) 

```

Comment:
If this line is not used anymore, we can remove it as well, if we need to recover it, we can go through the git history